### PR TITLE
Upversion to 4.6.0, switch to Eclipse 2023-06 update sites

### PR DESCRIPTION
**Pull request by olkornii**
Signed-off-by: Oleksii Korniienko <olkornii@redhat.com>

# Pull Request Checklist

## Code
* Are method-/class-/variable-names meaningful?
* Are methods concise, not too long?
* Are catch blocks catching precise Exceptions only (no catch all)?
* Is copyright with correct year in new files?
* Is there a JavaDoc in every class with description of a class and an author(and issue if suitable)?
* Is there a link to github project issue? (if needed)

## Testing
* Does it work?
* Are the tests green on Jenkins? Is there a link to Jenkins build?

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -16,13 +16,13 @@
...
 	<packaging>pom</packaging>
 
 	<properties>
 		<tycho-version>2.7.5</tycho-version>
 		<tychoExtrasVersion>${tycho-version}</tychoExtrasVersion>
-		<eclipse-target-site>https://download.eclipse.org/releases/2023-03/</eclipse-target-site>
+		<eclipse-target-site>https://download.eclipse.org/releases/2023-06/</eclipse-target-site>
 		<swtbot-update-site>https://download.eclipse.org/technology/swtbot/snapshots/</swtbot-update-site>
         <!-- when making final release, switch from S to R latest update site - https://github.com/eclipse/reddeer/issues/2100 -->
         <eclipse-orbit-update-site>https://download.eclipse.org/tools/orbit/downloads/latest-R/</eclipse-orbit-update-site>
```
Comment (odockal) [inline pom.xml:28]: Please see commented eclipse orbit repo url. During the development you should be using `latest-S` and switch to `latest-R` before final milestone release.

**Review by odockal — CHANGES_REQUESTED** [review]
Also, please take a look at `site/category.xml`. there are defined dependencies used during reddeer installation into eclipse via p2 update site. See my previous commits for this changes.

File: pom.xml (hunk 1 above)
Comment (olkornii) [reply pom.xml]: Done

**Comment by olkornii** [comment]
testPR

**Review by odockal — APPROVED** [review]

**Pull request merged**